lib/mquery.js

```diff
@@ -3195,7 +3134,4 @@ Query.env = require('./env');
 Query.Collection = require('./collection');
 Query.BaseCollection = require('./collection/collection');
 Query.Promise = Promise;
-module.exports = exports = Query;
-
-// TODO
-// test utils
+module.exports = exports = new Proxy(Query, withoutNew);
```

Comment:
I would strongly prefer to not export a proxy here, that seems like an unnecessary perf impact.

A couple of alternatives that we can consider:

1) Implement a `class QueryBase { ... }`, and then create a function that inherits from that class:

```javascript
class Base {
  constructor(name) {
    this.name = name;
  }

  sayHi() {
    console.log(`Hello, ${this.name}`);
  }
}

function Child(name) {
  if (!(this instanceof Child)) {
    return new Child(name);
  }
  this.name = name;
}

Child.prototype = Object.create(Base.prototype);
Child.prototype.constructor = Child;

const c = Child('World');
c.sayHi();
```

2) Put this in a breaking change, and perhaps add a separate `mquery.new()` function that allows creating a query instance without using the `new` operator.

Reply:
I would not have anything against a major change. 
`mquery.new()` seems a bit unnecessary...? and abnormal? in that case we can just say: hey use `new Query()` instead?

Reply:
> I would strongly prefer to not export a proxy here, that seems like an unnecessary perf impact.

Is it really impacting the perf that much?